propose_sns.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+#
+# This fails if the user does not follow the instructions at the top of
+# ./example_sns_init.yaml.
+#
+# This is mostly copied from deploy_sns.sh.
```

Comment:
Could you please indicate which steps in the SNS lifecycle from the README this replaces? E.g., is open_sns_swap.sh also necessary after running propose_sns.sh?

Reply:
I wasn't sure if that was in scope for the ticket.

Also, I'm not totally sure what the answer is. Let me try to explain here as best I can, and we can try to figure it out together:

I think once 1-proposal is "done", the following steps would be replaced (by this, or something else):

* 2: deploy_sns.sh (obviously)
* 3: I think this would be replaced by something else, like give_nns_control.sh (AFAIK, this does not exist yet)
* 5: open_sns_swap.sh: Yes, this would be ripped out as well. As the name implies, the point of 1-proposal is for sns-ification to be done in a single proposal as much as possible. IIUC, giving control of the dapp to NNS is not something that a proposal can do, so that would still have to be a separate step.
* 7: I think we recently made this superfluous. Andre implemented auto finalization (a feature that EVERYONE except Johan intuitively believes should have existed from the beginning).

Reply:
> 3: I think this would be replaced by something else, like give_nns_control.sh (AFAIK, this does not exist yet)

In that case, we should remark that you can only use `sns propose` for testing SNS without any registered canisters. Or add a script that adds SNS-W as an additional controller of the registered canister and call that before running `propose_sns.sh` (if I understand the new workflow correctly).

Reply:
FWIW, I think it's fine if we keep this script side-by-side with whatever is described in the README.md file for now.

Later on, we should rewrite the README file in a much simpler way, removing all (legacy) 2-proposal scripts and instructions. But I would pay the cost of actually doing so after we deprecate the legacy approach (which is a matter of several weeks from now, AFAIK).

Reply:
I like what Arshavir is saying, because it means I do not have to make README changes in this PR 😁  OTOH, now that I look at the ticket again, it calls for README changes.

Another way is to put the 1-proposal instructions in another file, and link to it from the main README, as Arshavir suggested during yesterday's team meeting. I will go with this. Approach. Please, stand by.

Reply:
Imho, if we now add another branch into the README file (e.g., "you could use approach A or approach B), it would overcomplicate the instructions.

If we have a smooth way of testing the 1-prop flow, which works on both apple silicon and x86, then let's just rewrite the README with the new official flow. I personally don't which ticket's scope this would fall into.

Reply:
I have come up with a new file: draft_1_proposal_instructions.md

It incorporates the changes I described in my first reply.

> Or add a script that adds SNS-W as an additional controller

Done.

I'm going to consider this particular thread resolved. Any issues arising from the new file should be discussed in new threads.

Reply:
Yeah, let's not document two ways. I was thinking that the other file would just hold a draft. I scrapped the idea of linking to it. The corresponding content in the main README would be replaced later, once stuff is more settled, and more tested.